src/raftstore/store/snap.rs

```diff
@@ -1951,11 +1950,13 @@ impl SnapManager {
         Ok(())
     }
 
-    pub fn list_snap(&self) -> io::Result<Vec<(SnapKey, bool)>> {
+    // Return all snapshots which is idle not being used.
+    pub fn list_idle_snap(&self) -> io::Result<Vec<(SnapKey, bool)>> {
         let core = self.core.rl();
         let path = Path::new(&core.base);
         let read_dir = try!(fs::read_dir(path));
-        Ok(read_dir.filter_map(|p| {
+        // Remove the duplicate snap keys.
+        let set: HashSet<_> = read_dir.filter_map(|p| {
```

Comment:
One V2 snapshot have several sst files. Their snap keys need to be uniqued.
The code is equivalent to `Vec::sort()` and then `Vec::dedup()`.